pyro/infer/traceenum_elbo.py

```diff
@@ -29,6 +29,18 @@ def _check_shared_scale(scales):
                          "but found {} different scales.".format(len(scales)))
 
 
+def _check_model_guide_enumeration_constraint(model_enum_sites, guide_trace):
```

Comment:
This check may be overly conservative - we should work out the minimal version of the criterion at some point.